### PR TITLE
Add hyperfine dephasing rate to NoiseModel

### PR DESCRIPTION
**Pull request by HGSilveri**
This does not allow for simulations with dephasing in the qutrit state, it simply distinguishes the dephasing rate for hyperfine ground states from the existing dephasing rate.

### CODE REVIEW

File: pulser-core/pulser/backend/noise_model.py [hunk 1]
```diff
@@ -81,8 +81,11 @@ class NoiseModel:
             deviation of a normal distribution centered in 1.
         relaxation_rate: The rate of relaxation from the Rydberg to the
             ground state (in 1/µs). Corresponds to 1/T1.
-        dephasing_rate: The rate of a dephasing error occuring (in 1/µs).
-            Corresponds to 1/T2.
+        dephasing_rate: The rate of a dephasing occuring (in 1/µs) when a
+            Rydberg state is involved in the superposition. Corresponds
+            to 1/T2*.
+        hyperfine_dephasing_rate: The rate of dephasing occuring (in 1/µs)
+            between hyperfine ground states.
```
Comment (a-corni) [inline pulser-core/pulser/backend/noise_model.py:88]: What about harmonising the two definitions ?
```suggestion
        dephasing_rate: The rate of a dephasing occuring (in 1/µs) between
            Rydberg and ground states. Only used if Rydberg state is involved in the superposition. Corresponds
            to 1/T2*.
        hyperfine_dephasing_rate: The rate of dephasing occuring (in 1/µs)
            between hyperfine and ground states. Only used if hyperfine state is involved in the superposition.
```

File: pulser-simulation/pulser_simulation/hamiltonian.py [hunk 2]
```diff
@@ -116,8 +116,12 @@ def basis_check(noise_type: str) -> None:
         local_collapse_ops = []
         if "dephasing" in config.noise_types:
             basis_check("dephasing")
-            coeff = np.sqrt(config.dephasing_rate / 2)
-            local_collapse_ops.append(coeff * qutip.sigmaz())
+            rate = (
+                config.hyperfine_dephasing_rate
+                if self.basis_name == "digital"
```
Comment (a-corni) [inline pulser-simulation/pulser_simulation/hamiltonian.py:121]: Can you have
```suggestion
                if "digital" in self.basis_name:
```
That would help me in my PR :)

Comment (a-corni) [reply pulser-simulation/pulser_simulation/hamiltonian.py:121]: Forget about this, there is no need

**Review by a-corni — APPROVED** [review]
Looks good to me !

File: pulser-core/pulser/backend/noise_model.py (hunk 1 above)
Comment (HGSilveri) [reply pulser-core/pulser/backend/noise_model.py:88]: Well, `dephasing_rate` is also used in XY so saying it's between ground and Rydberg is not really accurate imo. I'll harmonise with a few changes

**Pull request merged**